app/components/channel_drawer/channels_list/switch_teams_button/switch_teams_button.js

```diff
@@ -59,6 +64,23 @@ export default class SwitchTeamsButton extends React.PureComponent {
             />
         );
 
+        let teamIconContent;
+        if (lastTeamIconUpdate) {
+            const teamIconUrl = Client4.getTeamIconUrl(currentTeamId, lastTeamIconUpdate);
+            teamIconContent = (
+                <Image
+                    source={{uri: teamIconUrl}}
+                    style={styles.switcherTeamIconImage}
+                />
+            );
+        } else {
+            teamIconContent = (
+                <Text style={styles.switcherTeamIconText}>
+                    {displayName.substr(0, 2).toUpperCase()}
+                </Text>
+            );
+        }
```

Comment:
it looks like this block here is used in very places, perhaps it would be a good idea to create a component for it?

Reply:
@enahum yes you are right. I'll take a look and try to move it into a component